wallet-new/src/Cardano/Wallet/WalletLayer/Kernel/Wallets.hs

```diff
@@ -148,3 +152,21 @@ getWallets :: Kernel.DB -> IxSet V1.Wallet
 getWallets db = IxSet.fromList . map (toWallet db) . IxSet.toList $ allRoots
   where
     allRoots = db ^. dbHdWallets . HD.hdWalletsRoots
+
+-- | Gets Utxos per account of a wallet.
+getWalletUtxos
+    :: V1.WalletId
+    -> Kernel.DB
+    -> Either GetUtxosError [(V1.Account, Utxo)]
+getWalletUtxos wid db = runExcept $ do
+    accounts <- withExceptT GetUtxosErrorFromGetAccountsError $
+        exceptT $ getAccounts wid db
+
+    forM (IxSet.toList accounts) $ \account -> do
+        hdAccId <- withExceptT GetUtxosWalletIdDecodingFailed $
+            fromAccountId wid (V1.accIndex account)
```

Comment:
There is no need to use `fromAccountId` here. `account :: HdAccount`, so you can just use `account ^. hdAccountId`, so that you avoid this error condition.